test/unit/anchore_engine/services/test_api_specs.py

```diff
@@ -24,72 +24,33 @@
+    generated.
     """
 
     port = 8081
-
-    for swagger in service_swaggers:
-        pid = os.fork()
-        if not pid:
-            name = swagger.rsplit('/', 3)[2]
-            logger.info(('Starting server for: {} at: {}'.format(name, swagger)))
-            resolver = MockResolver(mock_all='all')
-            api_extra_args = {'resolver': resolver}
-
-            app = connexion.FlaskApp(name,
-                                     swagger_json=False,
-                                     swagger_ui=False)
-
-            app.add_api(swagger,
-                        resolver_error=True,
-                        validate_responses=True,
-                        strict_validation=True,
-                        **api_extra_args)
-
-            app.run(port=port)
-
-        else:
-            try:
-                logger.info('Wait for pinging server of pid: {}'.format(pid))
-                # Let the api initialize
-                retries = 3
-                killed = False
-
-                for i in range(retries):
-                    time.sleep(2)
-                    proc_id, status = os.waitpid(pid, os.WNOHANG)
-
-                    logger.info('Child pid: {}. Status = {}'.format(pid, status))
-                    if proc_id == 0 and status == 0:
-                        try:
-                            logger.info('Killing pid {}'.format(pid))
-                            os.kill(pid, signal.SIGTERM)
-                            killed = True
-                        except ProcessLookupError:
-                            logger.info('Process {} not found, skipping kill'.format(pid))
-                        finally:
-                            break
-                    elif status != 0 and killed:
-                        logger.info('Confirmed child pid killed')
-                        break
-                    else:
-                        pytest.fail('Mock service for {} failed to start properly'.format(swagger))
-                        break
-            except ProcessLookupError:
-                logger.info('Process {} not found. Exiting cleanly')
-            except (KeyboardInterrupt, Exception):
-                raise
-            finally:
-                try:
-                    os.kill(pid, signal.SIGKILL)
-                except ProcessLookupError:
-                    # This is expected
-                    pass
-                except:
-                    logger.exception("Failed to kill child process")
-
+    name = service.rsplit('/', 3)[2]
+    resolver = MockResolver(mock_all='all')
+    api_extra_args = {'resolver': resolver}
+
+    options = {"serve_spec": False, "swagger_ui": False}
+    app = connexion.FlaskApp(name, options=options)
+
+    app.add_api(service,
+                resolver_error=True,
+                validate_responses=True,
+                strict_validation=True,
+                **api_extra_args)
+
+    client = app.app.test_client()
```

Comment:
Does this fully test the spec itself? The objective of this test was to identify problems in the swagger spec (beyond just syntax linting) that the startup of the server identifies.

Specifically, things like references not resolving properly, missing required elements for a route spec, etc.

As long as those are covered with this usage, I'm happy with the change since it is certainly simpler.

Reply:
I was able to fiddle around with one of the YAMLs and see that an invalid reference was caught:

```text
test/unit/anchore_engine/services/test_api_specs.py::test_api_service[anchore_engine/services/apiext/swagger/swagger.yaml] FAILED
...
            try:
                document = document[part]
            except (TypeError, LookupError):
>               raise RefResolutionError(
                    "Unresolvable JSON pointer: %r" % fragment
                )
E               jsonschema.exceptions.RefResolutionError: Unresolvable JSON pointer: 'Ddefinitions/ApiErrorResponse'

../../.virtualenvs/anchore-engine/lib/python3.8/site-packages/jsonschema-2.6.0-py3.8.egg/jsonschema/validators.py:420: RefResolutionError
```

And I think this would also cover the missing required elements:

```text
E           connexion.exceptions.InvalidSpecification: Path parameter 'subscriptionId' for 'get' operation in '/subscriptions/{subscriptionId}' was not resolved
```